elastic-tube-1d/fluid-python/requirements.txt

```
@@ -0,0 +1,3 @@
+pyprecice~=3.0
+numpy
```

```suggestion
numpy >1, <2
```
See #564. Let's generally use this for numpy at the moment since numpy 2 is not widely supported yet.

---

We could also consider adding `numpy >1, <2` in `pyprecice` [here](https://github.com/precice/python-bindings/blob/1f750461c1d023f4de5d394804393be0566f904e/pyproject.toml#L3C1-L3C121). Then we would not need to specify the requirement here again. But I'm unsure what's the best practice here. Currently, we are duplicating the requirement but it also does not harm the way it is. I'm fine with either approach.

Assuming `pyprecice` might drop `numpy` at some point in time: If this case still needs `numpy` it would run into a problem. So I have a tendency towards keeping numpy here if there is an `import numpy` showing up anywhere.

---

I have not checked the details, but aren't we using the removed/changed functions also directly in the code we have in the tutorials? Then, specifying this as a restriction in the `pyprecice` would not be the right place.

Let's specify the exact restrictions here and remove the restrictions gradually, as things become compatible.